azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,363 @@
+
+        # get file extension
+        file_name = os.path.basename(file)
+        # get cert name by removing extension and path in front
+        cert_name = file_name.split(".")[0].replace(".", "")
+        cert_extension = self._validate_file_extension(file_name, ["der", "crt"])
+
+        secret_name = secret_name if secret_name else f"{cert_name}-{cert_extension}"
+
+        # iterate over secrets to check if secret with same name exists
+        secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+        self._upload_to_key_vault(secret_name, file, cert_extension)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        self._add_secret_to_spc(
+            secret_name=secret_name,
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+        )
+
+        # check if there is a secret sync called "aio-opc-ua-broker-trust-list ", if not create one
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        return self._add_secret_to_secret_sync(
+            secret_name=secret_name,
+            file_name=file_name,
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_TRUST_LIST_SECRET_SYNC_NAME,
+        )
+
+    def issuer_add(
+        self, instance_name: str, resource_group: str, file: str, secret_name: Optional[str] = None
+    ) -> dict:
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
+        secretsync_spc = self._find_existing_spc(instance_name=instance_name, cl_resources=cl_resources)
+        if not secretsync_spc:
+            return
+
+        # get properties from default spc
+        spc_properties = secretsync_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+
+        self.keyvault_client = get_keyvault_client(
+            subscription_id=self.default_subscription_id,
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # get file extension
+        file_name = os.path.basename(file)
+        # get cert name by removing extension and path in front
+        cert_name = file_name.split(".")[0].replace(".", "")
```

Comment:
use the os path library for pathing - you will not have a fun time with different oses

also what if the cert name has .'s in them?

Reply:
If you split the file_name, `file_name.split(".")[0]` will only return the first part of a file with `.` in the name:
```
>>> file_name
'file.test.txt'
>>> file_name.split(".")[0].replace(".", "")
'file'
```

I believe what we need here is just:
```suggestion
        cert_name = file_name.replace(".", "-")
```
Which gives us:
```
>>> file_name.replace(".", "-")
'file-test-txt'
```